### PR TITLE
Add new bot: cut string from string

### PR DESCRIPTION
**Pull request by mariuskarotkis**
Add new bot: cut string from string

### CODE REVIEW

File: intelmq/bots/experts/cut_from_string/expert.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+"""
+Cut from string
+"""
+from intelmq.lib.bot import Bot
+
+
+class CutFromStringExpertBot(Bot):
+    string_from_start: int = 1  # 1 - from start, 0 - from end
```
Comment (wagner-certat) [inline intelmq/bots/experts/cut_from_string/expert.py:9]: If the valid values for these parameters are just 0 and 1 why not use a boolean value? From a user-perspective, maybe string values like `cut_from = 'start'` are better? (just thinking aloud)

**Review by wagner-certat — CHANGES_REQUESTED** [review]

**Review by wagner-certat — CHANGES_REQUESTED** [review]
Please add documentation in `docs/user/bots.rst`

File: intelmq/bots/experts/cut_from_string/expert.py (hunk 1 above)
Comment (mariuskarotkis) [reply intelmq/bots/experts/cut_from_string/expert.py:9]: Good point.
It can be improved.

Comment (schacht-certat) [reply intelmq/bots/experts/cut_from_string/expert.py:9]: Given that there are only two possible values, it should IMO be a boolean

File: intelmq/bots/experts/cut_from_string/expert.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    string_for_cut: str = 'www.'
+    field_for_cut: str = 'source.fqdn'
+
+    def init(self):
+        pass
+
```
Comment (schacht-certat) [inline intelmq/bots/experts/cut_from_string/expert.py:15]: If this method doesn't do anything, we maybe simple can remove it?

File: intelmq/bots/experts/cut_from_string/expert.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                field_string = field_string[len(self.string_for_cut):]
+                event.change(self.field_for_cut, field_string)
+
+            if self.string_from_start == 0 and field_string.endswith(self.string_for_cut):
+                field_string = field_string[:-len(self.string_for_cut)]
+                event.change(self.field_for_cut, field_string)
+
+        self.send_message(event)
+        self.acknowledge_message()
+
```
Comment (schacht-certat) [inline intelmq/bots/experts/cut_from_string/expert.py:31]: PEP616 lists example code for the `removeprefix` and `removesuffix` String methods- maybe we can use those here and add a comment to drop them once we only support Python versions that implement those methods:
```
def removeprefix(self: str, prefix: str, /) -> str:
    if self.startswith(prefix):
        return self[len(prefix):]
    else:
        return self[:]

def removesuffix(self: str, suffix: str, /) -> str:
    # suffix='' should not call self[:-0].
    if suffix and self.endswith(suffix):
        return self[:-len(suffix)]
    else:
        return self[:]
```

then the code would look like this:
```
if self.removeprefix:
    event.change(self.field, self.removeprefix(self.field, self.affix)
else:
    event.change(self.field, self.removesuffix(self.field, self.affix)
```

File: intelmq/bots/experts/cut_from_string/expert.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+"""
+Cut from string
+"""
+from intelmq.lib.bot import Bot
+
+
+class CutFromStringExpertBot(Bot):
+    string_from_start: int = 1  # 1 - from start, 0 - from end
+    string_for_cut: str = 'www.'
+    field_for_cut: str = 'source.fqdn'
```
Comment (schacht-certat) [inline intelmq/bots/experts/cut_from_string/expert.py:11]: I think `field_for_cut` should simply be called `field` - there are already some other bots that operate on fields, that use the variable name `field`

**Review by schacht-certat — CHANGES_REQUESTED** [review]
If I understand correctly, the bot will remove a prefix or a suffix from a field. So I would call it RemoveAffix. There is also [PEP616](https://www.python.org/dev/peps/pep-0616/) which proposes `str` methods which do exactly that and which seems to have been [accepted](https://bugs.python.org/issue39939) for Python 3.9. So once we only support Python 3.9, we can use that (maybe add a comment so that we don't forget).

Please add a reuse compliant license and copyright header to the files, an example can be found in the [documentation](https://intelmq.readthedocs.io/en/latest/dev/guide.html#template)

File: intelmq/bots/experts/cut_from_string/expert.py (hunk 2 above)
Comment (mariuskarotkis) [reply intelmq/bots/experts/cut_from_string/expert.py:15]: I will remove it.

**Comment by mariuskarotkis** [comment]
Updated

**Comment by mariuskarotkis** [comment]
> If I understand correctly, the bot will remove a prefix or a suffix from a field. So I would call it RemoveAffix. There is also [PEP616](https://www.python.org/dev/peps/pep-0616/) which proposes `str` methods which do exactly that and which seems to have been [accepted](https://bugs.python.org/issue39939) for Python 3.9. So once we only support Python 3.9, we can use that (maybe add a comment so that we don't forget).
> 
> Please add a reuse compliant license and copyright header to the files, an example can be found in the [documentation](https://intelmq.readthedocs.io/en/latest/dev/guide.html#template)

Please check.

**Comment by ghost** [comment]
I think the only discussion point left open is this comment/proposal by @schacht-certat:
> If I understand correctly, the bot will remove a prefix or a suffix from a field. So I would call it RemoveAffix.

@mariuskarotkis What do you think of this idea? Does something indicate against it?

**Comment by mariuskarotkis** [comment]
> RemoveAffix

It's good for me. I think it can be renamed.

**Review by wagner-certat — APPROVED** [review]

**Comment by ghost** [comment]
Thank you very much - again! - for this useful addition.

**Pull request merged**